### ISSUE TITLE
Hubspot Contact update on Posthog Event

### ISSUE THREAD

**Issue by oneshot-engineering** [PostHog/hubspot-plugin#13]
Can we please make Hupspot plugin to support both update or insert onEvent. Looks like it only creates hubspot contact as of now. We have some use case around marketing campaign to capture sign up much prior to app sign up. Once the same user come to app for signup, We just wanted to update Hubspot Contact.

**Comment by yakkomajuri** [comment]
@kpthatsme

**Comment by oneshot-engineering** [comment]
Hi @yakkomajuri  @kpthatsme 

If it's not too much ask, Do you think If it can be picked up anytime soon?

Just wanted to plan few things on our end which are bit dependent on this. It would be helpful for me to prioritize based on this.

**Comment by kpthatsme** [comment]
Hey @oneshot-engineering there isn't a timeline for this right now - at least not for the next week or 2. 

You're welcome to submit a pull-request with the functionality you want added!

**Comment by oneshot-engineering** [comment]
Ok thanks for letting me know. I haven't quickly looked into the code yet. How easy for this to test if my team pick it up? We use cloud version. If you have quick stand alone tests that my team can modify the code and run the tests. That should be good for my engineers to take on.

**Comment by kpthatsme** [comment]
hey @oneshot-engineering I just put a quick PR for this actually, wanna review for me?

https://github.com/PostHog/hubspot-plugin/pull/14


Full disclosure I haven't tested this but I think this approach should work

**Comment by oneshot-engineering** [comment]
Perfect sounds good.  Let me take a look

**Comment by oneshot-engineering** [comment]
Nice, It looks good @kpthatsme.

You are just updating the contact whenever create is failed. Should satisfy the use case.

**Comment by kpthatsme** [comment]
Just merged it in @oneshot-engineering - if you wanna give it a shot, update your plugin in the app. 

Note: I still haven't tested it yet

**Comment by oneshot-engineering** [comment]
update plugin as in? Am in cloud version, is this deployed?

**Comment by kpthatsme** [comment]
Yes it's merged, if you go into the plugin page in app - click "check for updates" or something like that near the top of your installed plugins. 

Then you'll see an update available for this plugin that you can grab.

**Comment by oneshot-engineering** [comment]
It says this.  "All your plugins are up to date. Great work!". So hope the plugin is updated. 

I have this configured in the hubspot plugin and onEvent is $identify.
created_at:ph_signup_date

I was checking for hubspot contacts for the latest identify events and i don't see the hubspot property (ph_signup_date) is getting updated though.

**Comment by oneshot-engineering** [comment]
@kpthatsme  - Any update on this?

**Comment by kpthatsme** [comment]
Hey @oneshot-engineering sorry for the delay. It looks like Hubspot changed their API spec for updating contacts. We pushed a fix earlier today and things look good on our end. 

Could you try updating your plugin (same instructions as before) and refresh your plugin page (to clear old error messages) ? Let us know if that works!
